sdks/java/core/src/main/java/org/apache/beam/sdk/util/WindowedValue.java

```diff
@@ -59,16 +64,91 @@
 @Internal
 public abstract class WindowedValue<T> {
 
+  public interface Outputter<T> {
+    void output(OutputBuilder<T> outputBuilder);
+  }
+
+  /** Create a typical {@link OutputBuilder} that */
+  public static <T> OutputBuilder<T> builder(Outputter<T> outputter) {
+    return new AutoBuilder_WindowedValue_Builder<T>().setOutputter(outputter);
+  }
+
+  @AutoBuilder(callMethod = "of", ofClass = WindowedValue.class)
```

Comment:
I have to say this really only saves like a couple lines of code, so the `Auto` part is not very important.